### PR TITLE
IEEE-126-Convert-TeamCard-to-TypeScript and IEEE-127-Connect-TeamCard-on-dashboard-to-redux-store

### PR DESCRIPTION
**Pull request by Mustaballer**
## Overview

- Resolves IEEE-126 and IEEE-127
- Converted TeamCard.js, TeamCard.test.js, Team.js, Team.test.js to TypeScript
- Connected TeamCard on Dashboard to redux store

## Unit Tests Created

- check to see team member names and team code is rendered

## Steps to QA

- go to the dashboard
- make sure you have a team
- add users to your team --> notice their names will show up

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import TeamCard from "components/dashboard/TeamCard/TeamCard";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import { fireEvent } from "@testing-library/react";
+import Cart from "../../../pages/Cart/Cart";
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: absolute imports pleasseeee

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import Teams from "../../../pages/Teams/Teams";
+
+describe("TeamCard", () => {
+    const members = ["Aki", "Graham", "Alex", "Raymond"];
+    const teamCode = "PAS3NLQ3";
+    test("place holder test", () => {
+        expect(teamCode).not.toBeNull();
+    });
+
+    // test("render", () => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: uncomment & update these tests

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx [hunk 3]
```diff
@@ -2,27 +2,28 @@ import React from "react";
...
         <TitledPaper title={title}>
-            {members.map((member, i) => (
+            {members?.map((member, i) => (
                 <Container className={styles.name} key={i}>
                     <Typography variant="body2" noWrap>
                         {member}
                     </Typography>
                 </Container>
             ))}
-
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx]: we don't need to delete this

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx [hunk 4]
```diff
@@ -77,6 +84,25 @@ it("Opens Product Overview with the correct hardware information", async () => {
             expect(getByText(hardware.model_number)).toBeInTheDocument();
             expect(getByText(hardware.manufacturer)).toBeInTheDocument();
             expect(getByText(hardware.notes)).toBeInTheDocument();
+            //
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx]: remove this line

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx [hunk 5]
```diff
@@ -11,20 +11,25 @@ import {
...
+    getCurrentTeam,
+    isLoadingSelector,
+    teamSelector,
+} from "slices/event/teamSlice";
+import LinearProgress from "@material-ui/core/LinearProgress";
 
 const Dashboard = () => {
     const dispatch = useDispatch();
+    const currentTeam = useSelector(teamSelector);
+    const memberNames = currentTeam?.profiles.flatMap(
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx]: let's take advantage of redux and make a selector for the team code and member list

**Review by YuYing-Liang — CHANGES_REQUESTED** [review]
Just some initial thoughts, tested it out and looks good!

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx (hunk 5 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx]: sounds good!

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx (hunk 2 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: @YuYing-Liang For the handle edit team test, can I remove that one since we are not using the edit button to change the teams

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx (hunk 3 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx]: should we keep the handleEditTeam as a prop? @YuYing-Liang

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx (hunk 2 above)
Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: keep it for now because we'll use it later, Leo's working on a ticket for the edit modal currently.

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx (hunk 3 above)
Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx]: Yupp let's keep it

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx (hunk 2 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: can I remove the snapshot test, cause we got rid of the snapshots? @YuYing-Liang

Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: Yup go for it

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx [hunk 6]
```diff
@@ -1,18 +1,14 @@
...
 describe("TeamCard", () => {
     const members = ["Aki", "Graham", "Alex", "Raymond"];
     const teamCode = "PAS3NLQ3";
-
-    test("render", () => {
-        const { asFragment } = render(
-            <TeamCard members={members} teamCode={teamCode} />
-        );
-        expect(asFragment()).toMatchSnapshot();
+    test("place holder test", () => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: what's the point of this test?

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx [hunk 7]
```diff
@@ -2,22 +2,30 @@ import React from "react";
...
-const TeamCard = ({ members, teamCode, handleEditTeam }) => {
-    const title = teamCode === "" ? "Team" : "Team " + teamCode;
+interface TeamProps {
+    members?: string[];
+    teamCode?: string;
+    handleEditTeam?: any;
+}
 
+const TeamCard = ({ members, teamCode, handleEditTeam }: TeamProps) => {
+    const title = teamCode === "" ? "Team" : "Team " + teamCode;
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx]: is the teamCode empty string or null? It should be null so might need to change how its updated in the slice.
also, get rid of the members and teamCode props, since we're using redux, we should get those values from the slice selectors

**Review by YuYing-Liang — CHANGES_REQUESTED** [review]
Just some small things, sorry for not catching this last time!

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx (hunk 6 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.test.tsx]: yeah I can't remember lol

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx [hunk 8]
```diff
@@ -2,24 +2,39 @@ import React from "react";
...
+    handleEditTeam?: any;
+}
 
+export const TeamCard = ({ handleEditTeam }: TeamProps) => {
+    const members = useSelector(teamMemberNamesSelector);
+    const teamCode = useSelector(teamCodeSelector);
+
+    const title = teamCode === null ? "Team" : "Team " + teamCode;
+    // waiting for the team store
+    if (!members) {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx]: could you instead do members.length ? members?.map(...) : null in the return statement 
Only suggesting because we don't usually have 2 returns for a component

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx [hunk 9]
```diff
@@ -2,24 +2,39 @@ import React from "react";
...
-    const title = teamCode === "" ? "Team" : "Team " + teamCode;
+interface TeamProps {
+    handleEditTeam?: any;
+}
 
+export const TeamCard = ({ handleEditTeam }: TeamProps) => {
+    const members = useSelector(teamMemberNamesSelector);
+    const teamCode = useSelector(teamCodeSelector);
+
+    const title = teamCode === null ? "Team" : "Team " + teamCode;
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx]: Can we change this to if you don't have a team it should say "No Team"
And add a test for this (for when teamCode is null)

File: hackathon_site/dashboard/frontend/src/slices/event/teamSlice.ts [hunk 10]
```diff
@@ -96,3 +95,16 @@ export const teamSelector = createSelector(
     [teamSliceSelector],
     (teamSlice) => teamSlice.team
 );
+
+export const teamCodeSelector = createSelector(
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/event/teamSlice.ts]: you need to add tests to the teamslice.test.ts for these selectors. should be a small change

**Review by YuYing-Liang — CHANGES_REQUESTED** [review]
Missed some stuff again, sorry! Promise these are the last changes

File: hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+            {members?.length
+                ? members?.map((member, i) => (
+                      <Container className={styles.name} key={i}>
+                          <Typography variant="body2" noWrap>
+                              {member}
+                          </Typography>
+                      </Container>
+                  ))
+                : null}
+            <Container className={styles.lastRow}>
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/TeamCard/TeamCard.tsx:31]: I will tell Leo to change this in his PR bc I've requested so many changes already, but this button should be disabled if there's no team

**Review by YuYing-Liang — APPROVED** [review]
Looks good to me!
